frontend/packages/kubevirt-plugin/integration-tests/tests/models/virtualMachine.ts

```diff
@@ -28,8 +28,8 @@ import { KubevirtDetailView } from './kubevirtDetailView';
 import { ImportWizard } from './importWizard';
 
 export class VirtualMachine extends KubevirtDetailView {
-  constructor(config) {
-    super({ ...config, kind: 'virtualmachines' });
+  constructor(config, kind?: 'virtualmachines' | 'virtualmachineinstances') {
```

Comment:
> whether we should or should not differentiate the type of VirtualMachine and VirtualMachineInstance more explicitly.

The page rendering VM and VMI share the same component, it renders a `vmiLike` object that can be VM or VMI. When rendering we have small changes, mainly page title, how status is calculated, and what is allowed to be edit:
https://github.com/openshift/console/blob/master/frontend/packages/kubevirt-plugin/src/components/vms/vm-details.tsx#L72

IMHO it makes sense to share the testing methods of this components too.

Reply:
VM and VMI objects are very similar, not the same though. You can't do the same (or any basically) actions with VMI, which would be nice to have reflected in our testing framework. 
But it could be achieved this way as well.

Reply:
yes there will be differences in actions and also editing and we should reflect that and not use the same class for them. Let's figure out the best approach

Reply:
ok; we decided to keep it simple for now after offline discussion. 

Radim will revisits this once his VMBuilder PR is up.

Let's just think of it as temporary solution for now.